lib/auth_web/controllers/auth_controller.ex

```diff
@@ -189,6 +211,19 @@ defmodule AuthWeb.AuthController do
 
     # email is blank or invalid:
     if is_nil(email) or not Fields.Validate.email(email) do
+      # intialise login log data
+      user_agent = get_user_agent(conn)
```

Comment:
if these variables aren't used anywhere else, they could be inlined in the `login_log` Map below. 💭